### PR TITLE
feat: extract from path

### PR DESCRIPTION
**Pull request by baszalmstra**
Functions to extract packages from a path.

### CODE REVIEW

**Review by wolfv — APPROVED** [review]

**Comment by wolfv** [comment]
looks super nice.

File: crates/rattler_package_streaming/src/fs.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+use crate::{ArchiveType, ExtractError};
+use std::fs::File;
+use std::path::Path;
+
+/// Extracts the contents a `.tar.bz2` package archive from the specified remote location.
```
Comment (wolfv) [inline crates/rattler_package_streaming/src/fs.rs:5]: remote location? I interpret the function as taking a `fs::path`?

File: crates/rattler_package_streaming/src/fs.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+///     Path::new("/tmp"))
+///     .unwrap();
+/// ```
+pub fn extract_conda(archive: &Path, destination: &Path) -> Result<(), ExtractError> {
+    let file = File::open(archive)?;
+    crate::read::extract_conda(file, destination)
+}
+
+/// Extracts the contents a package archive from the specified remote location. The type of package
+/// is determined based on the path of the url.
```
Comment (wolfv) [inline crates/rattler_package_streaming/src/fs.rs:36]: based on the file extension of the path?

File: crates/rattler_package_streaming/src/fs.rs (hunk 1 above)
Comment (baszalmstra) [reply crates/rattler_package_streaming/src/fs.rs]: Changed

File: crates/rattler_package_streaming/src/fs.rs (hunk 2 above)
Comment (baszalmstra) [reply crates/rattler_package_streaming/src/fs.rs]: Changed

**Pull request merged**